src/main/java/uk/gov/digital/ho/pttg/application/retry/NameMatchingRetryPolicy.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.digital.ho.pttg.application.retry;
+
+import org.springframework.retry.RetryContext;
+import uk.gov.digital.ho.pttg.application.ApplicationExceptions;
+import uk.gov.digital.ho.pttg.dto.Individual;
+
+import java.util.List;
+
+public class NameMatchingRetryPolicy extends ExceptionAwareRetryPolicy {
```

Comment:
Not sure if retry policy is the right approach for retrying the names. It's introduced some state and seems complex and could quite slow. This could be achieved by looping/streaming over the candidate names?